src/Bicep.Core/Intermediate/ExpressionBuilder.cs

```diff
@@ -882,6 +888,14 @@ private Expression ConvertVariableAccess(VariableAccessSyntax variableAccessSynt
             case LocalVariableSymbol localVariableSymbol:
                 return GetLocalVariableExpression(variableAccessSyntax, localVariableSymbol);
 
+            case ImportedSymbol importedSymbol:
+                if (importedSymbol.Kind == SymbolKind.Variable)
+                {
+                    return new ImportedVariableReferenceExpression(variableAccessSyntax, importedSymbol);
+                }
+
+                throw new InvalidOperationException($"Cannot dereference an imported symbol of kind {importedSymbol.Kind} in a value expression.");
```

Comment:
Presumably caught by earlier type validation? Could you add a comment to explain this and ensure we have a test to cover it?